js/renovation/component_wrapper/utils/update_props_immutable.ts

```diff
@@ -24,7 +25,11 @@ function cloneObjectProp(
       fullNameParts.slice(1),
     );
   } else if (name) {
-    result[name] = value[name];
+    if (isPlainObject(value[name])) {
+      result[name] = extend(true, {}, value[name]);
```

Comment:
Deep extend is bad. Why is it necessary? Where is jest tests for it? May be one level extend is enough?
```javascript
result[name] =  { ...value[name] };
```

Reply:
In my case one level extend is enough, but it might be not enough in some other cases.

The problem was in this case (it's test called `Change toolbar.items[i] in runtime` in `dataGrid.tests.js`):

- set debugger to `js\renovation\ui\grids\data_grid\utils\get_updated_options.ts:28` (after `else if (type1 === 'array')`)
- I have toolbar with toolbar with two buttons, both are `before`
```
toolbar: { items: [  {
    name: 'columnChooserButton',
    location: 'before'
}, {
    name: 'addRowButton',
    location: 'before'
} ] }
```
- I'm changing second button: `grid.option('toolbar.items[1]', { name: 'addRowButton', location: 'after' });`
- In debugger `item1 !== item2` is true, but their content is equal: both have `location: 'after'` for second button, so `optionChanged` won't be triggered. It happens because `item1[1] === item2[1]`

I'm not sure if one level extend will work in other cases, but i can't give an example yet